atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/ValueStoreImpl.java

```diff
@@ -51,6 +52,7 @@ final class ValueStoreImpl implements ValueStore {
 
     private final StructureHolder<io.vavr.collection.Map<CellReference, CacheEntry>> values;
     private final StructureHolder<io.vavr.collection.Set<TableReference>> watchedTables;
+    private final StructureHolder<io.vavr.collection.Set<RowReference>> watchedRows;
```

Comment:
This could theoretically get quite large (there's a lot more rows out there than tables). Should we add a sensible limit to the amount of rows that can be watched at once?